utils/s3/client.cc

```diff
@@ -470,13 +470,16 @@ class client::readable_file : public file_impl {
     }
 
     virtual future<struct stat> stat(void) override {
-        auto size = co_await _client->get_object_size(_object_name);
+        auto object_stats = co_await _client->get_object_stats(_object_name);
         struct stat ret {};
         ret.st_nlink = 1;
         ret.st_mode = S_IFREG | S_IRUSR | S_IRGRP | S_IROTH;
-        ret.st_size = size;
+        ret.st_size = object_stats.size;
         ret.st_blksize = 1 << 10; // huh?
-        ret.st_blocks = size >> 9;
+        ret.st_blocks = object_stats.size >> 9;
```

Comment:
despite that this is not part of this change, this still caught my eye. if we consider 1K as the block size, then the number of blocks should be `size >> 10`. @xemul what do you think?

Reply:
IIRC, st_blksize and st_blocks are unrelated. The former refers to preffered block size, for I/O efficiency. The latter applies system-specific block size. On Linux, always 512 bytes.

Reply:
Yes, see https://pubs.opengroup.org/onlinepubs/009696799/basedefs/sys/stat.h.html
```
The unit for the st_blocks member of the stat structure is not defined within IEEE Std 1003.1-2001. In some implementations it is 512 bytes. It may differ on a file system basis. There is no correlation between values of the st_blocks and st_blksize, and the f_bsize (from [<sys/statvfs.h>](https://pubs.opengroup.org/onlinepubs/009696799/basedefs/sys/statvfs.h.html)) structure members.
```

Reply:
thanks. that makes more sense now.